core/domain/entities/routing/handlers/admin/WordPressPluginsPage.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace EventEspresso\core\domain\entities\routing\handlers\admin;
+
+use EE_Dependency_Map;
+use EventEspresso\core\domain\entities\routing\handlers\Route;
+use EventEspresso\core\domain\services\admin\PluginUpsells;
+
+/**
+ * Class WordPressPluginsPage
+ * detects and executes logic for the WordPress Plugins admin page
+ *
+ * @package EventEspresso\core\domain\entities\routing\handlers\admin
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class WordPressPluginsPage extends Route
+{
+
+    /**
+     * returns true if the current request matches this route
+     *
+     * @return bool
+     * @since   $VID:$
+     */
+    public function matchesCurrentRequest()
+    {
+        return $this->request->isAdmin()
+               && strpos($this->request->requestUri(), 'wp-admin/plugins.php') !== false;
```

Comment:
Is this always guaranteed? What about the folks who rename `wp-admin` to something else to protect it from being hammered?

Reply:
I think a better way may be to use [WP Screen](http://codex.wordpress.org/Class_Reference/WP_Screen) API

Reply:
oh great point.. ya will fix

Reply:
couldn't use `WP_Screen` because that is created too late, but I still fixed the issue

Reply:
But here it's still the same, may be you forgot to make changes to this file?

Reply:
DOH... i fixed the route match specification but not this class..